### ISSUE TITLE
impossible to create document in public shared directory

### ISSUE THREAD

**Issue by pangea-org** [nextcloud/richdocuments#1581]
**Describe the bug**
in a shared directory with permission to upload and edit it is impossible to create collabora managed documents.

### Steps to reproduce
1. Create a shared directory with upload and edit permission.
2. Open the directory as guest (not a logged in user).
3. Create a collabora file (.odt, .ods, .odp) pushing the '+' button

### Expected behaviour
a new rich text document is created in the shared directory.

### Actual behaviour
Nothing happen, the file is not created and no error messages are displayed.

### Server configuration
**Operating system**: debian 10.8

**Web server:** Apache/2.4.38 

**Database:** MariaDB 10.3.17 

**PHP version:** 7.3

**Nextcloud version:** 20.0.7

**Version of Collabora Online**   3.8.2

<details>
<summary>Web server error log</summary>
```
"GET /ocs/v2.php/apps/richdocuments/api/v1/templates/document HTTP/2.0" 401 598
```
</details>

**Comment by mmallejac** [comment]
I am having the same exact issue.

I've tested that with Nextcloud 20.0.10 and 21.0.2 (docker or metal), with richdocument app versions 3.8.2 and 4.1
The behavior is the same when the shared folder is within a groupfolder or a standard folder.

**Comment by unteem** [comment]
Hi. We are also facing this issue with version 3.8.2 and version 4.1.1
Looks like its also in version 3.7.17 https://github.com/nextcloud/richdocuments/issues/622#issuecomment-815792893

Each time with CODE v6. 4

So when an unregistered user try to create an office file in publicly shared folder, he/she gets a 401 error. 

A workaround is for that user to first upload a document, he/she can then edit it in collabora.

@juliushaertl sorry for the ping but it looks like a regression from a previous fix you made, if you can have a look or point us in the right direction maybe we could help. Thanks.

**Comment by juliushaertl** [comment]
I can reproduce it, seems to be an issue where it tries to load templates from an endpoint that is only supposed to be called by logged in users.

**Comment by eBug** [comment]
This is also an issue with NC 23.0.0 and Collabora Office 6.4-50 (git hash: d53ee72), when I share a public link to a folder, anonymous users can't create or edit documents online.

**Comment by sergeng** [comment]
> This is also an issue with NC 23.0.0 and Collabora Office 6.4-50 (git hash: d53ee72), when I share a public link to a folder, anonymous users can't create or edit documents online.

NC 23.0.1RC
Richdocuments is whitelisted in Guest app
yet exact same issue !

**Comment by eneiluj** [comment]
@eBug @sergeng Have you tried with the Nextcloud Office app v5.0.1? I'm pretty sure file creation in public folder shares was fixed.

**Comment by eneiluj** [comment]
It appears to have been fixed after v5.0.1 actually. Next release will fix this issue then.

**Comment by juliushaertl** [comment]
The fix was published with 5.0.2 now.
